### ISSUE TITLE
More logging/warnings and more configurable logging/warnings

### ISSUE THREAD

**Issue by oxinabox** [JuliaML/MLDataPattern.jl#13]
While playing with #9,
I missed the number of factors my total number of observations had.
I set the `maxsize` to `100_000`,
and the actual size ended up at `4`.
Which given I was transferring data 1 minibatch at a time to the TensorFlow backend,
causes problems, given inter-process communication overheads.
So in this case, I think that there should be some cut-off, for use of `upto` sizing,
where a warning is given if the eventual size is (for example) less than `10%` the max size.

In my problem, it turned out using huge batches causes training problems, making it slow to learn.
Possibly really effective solution is to actually do some/all of the batching inside TensorFlow. But it works out fine to just reduce the batch size to 16,000 or so - using fixed size, since there are no divisors near that.
But then I have a remainder.

Each epoch, I am shuffling the data then re-cutting my batchs -- so that the points getting skipped change each time.
This means I get a message every epoch:
```
INFO: The specified values for size and/or count will result in 15780 unused data points
```

Which is annoying.
And if more logging is added for eg `maxsize` then this is going to get more annoying.

I suggest some kind of logging support is implemented.
Possibly by setting a global log-level variable (rather than passing around a parameter everywhere).

I suggest controls should allow:

 - print every time *(current)*
 - print once per message  (just store a set of log messages, and check if this message has been printed before)
 - print once per call-site (I think this can be done in a function by examining the call stack at runtime, but I am not sure. It might need a macro, to capture the @__LINE__).
 - print once per message, per call-site

This might have to wait until better logging support for Base.:
https://github.com/JuliaLang/Juleps/pull/30

**Comment by Evizero** [comment]
> warning is given if the eventual size is (for example) less than 10% the max size

Good idea.

> This means I get a message every epoch:

Yes we should have a generic way to surpress this. Maybe introduce a logging package

> print once per call-site 

would be possible with `Base.depwarn` already

**Comment by Evizero** [comment]
fixed on master by switching to `Base.warn_once`
